tests/Tests.AzureAppConfiguration/MockTestData/appconfig-settings.json

```diff
@@ -0,0 +1,132 @@
+[
+  {
+    "key": "key01",
+    "value": "Some Text",
+    "contentType": "application/json"
+  },
+  {
+    "key": "key02",
+    "value": true,
+    "contentType": "application/json;"
+  },
+  {
+    "key": "key03",
+    "value": null,
+    "contentType": "application/activity+json"
+  },
+  {
+    "key": "key04",
+    "value": "false",
+    "contentType": "application/json+text;"
+  },
+  {
+    "key": "key05",
+    "value": "null",
+    "contentType": "application/json; charset=utf-8;"
+  },
+  {
+    "key": "key06",
+    "value": 23.4,
+    "contentType": "application/json;charset=utf-8;param=value"
+  },
+  {
+    "key": "key07",
+    "value": [ 1, 2, 3 ],
+    "contentType": "application/vnd.dummy+json+foobar"
+  },
+  {
+    "key": "key08",
+    "value": [ "abc", "def", "ghi" ],
+    "contentType": "application/vnd.microsoft.appconfig+json;charset=utf-8"
+  },
+  {
+    "key": "key09",
+    "value": [],
+    "contentType": "application/json"
+  },
+  {
+    "key": "key10",
+    "value": [
+      1000,
+      true,
+      "AppConfig string value",
+      [ "NestedArray", 12, false, null ],
+      {
+        "ObjectSetting": {
+          "Logging": {
+            "LogLevel": "Information",
+            "Default": "Debug"
+          }
+        }
+      }
+    ],
+    "contentType": "application/json"
+  },
+  {
+    "key": "key11",
+    "value": [
+      [ "NestedArray4", 34 ],
+      [ "NestedArray5", true, null ],
+      [
+        "NestedArray6",
+        {
+          "MyNullValue": null,
+          "MyObject": {
+            "Property": {
+              "Name": {
+                "Name1": "Value1",
+                "Name2": "Value2"
+              }
+            }
+          }
+        }
+      ]
+    ],
+    "contentType": " application/foo+bar+json "
+  },
+  {
+    "key": "key12",
+    "value": {
+      "ObjectSetting": {
+        "Logging": {
+          "Default": "Debug",
+          "LogLevel": "Information"
+        }
+      }
+    },
+    "contentType": "application/json"
+  },
+  {
+    "key": "key13",
+    "value": {
+      "Obj1": null,
+      "Obj2": false,
+      "Obj3": 666
+    },
+    "contentType": "application/json"
+  },
+  {
+    "key": "key14",
+    "value": {},
+    "contentType": "application/json"
+  },
+  {
+    "key": "key15",
+    "value": "",
+    "contentType": "application/json"
+  },
+  {
+    "key": "key16",
+    "value": [
+      [ "NestedArray1", "12", "false", "null" ],
+      [ "NestedArray2", "text" ],
+      [ "NestedArray3" ]
+    ],
+    "contentType": "application/json"
+  },
+  {
+    "key": "key17",
+    "value": "2020-01-01T12:34:56.789Z",
+    "contentType": "application/json"
+  }
+]
```

Comment:
Can we add test cases for key-vault reference and feature flag content types? That will help validate those values remain unaffected, even though they are valid json types.

Reply:
I added a test for feature flags (JsonContentTypeTests_DontFlattenFeatureFlagAsJsonObject) when users don't configure "UseFeatureFlags". In that case, we should be able to read the value of feature flags as regular string (no object flattening).

I did not add tests for when users do configure Feature Flags and KeyVault refs because we already have explicit tests for these. As long as those tests pass, we don't need to add duplicate tests here. I think it's best to keep them decoupled in case we make changes to Feature Flags or KeyVault functionality in future.


---
In reply to: [453143374](https://github.com/Azure/AppConfiguration-DotnetProvider/pull/181#discussion_r453143374) [](ancestors = 453143374)

Reply:
Added more tests.

---
In reply to: [453921593](https://github.com/Azure/AppConfiguration-DotnetProvider/pull/181#discussion_r453921593) [](ancestors = 453921593,453143374)